velox/type/Type.cpp

```diff
@@ -835,13 +835,14 @@ RowTypePtr ROW(std::vector<TypePtr>&& types) {
 }
 
 MapTypePtr MAP(TypePtr keyType, TypePtr valType) {
-  return std::make_shared<MapType>(std::move(keyType), std::move(valType));
+  return TypeFactory<TypeKind::MAP>::create(
+      std::move(keyType), std::move(valType));
 }
 
 std::shared_ptr<const FunctionType> FUNCTION(
     std::vector<TypePtr>&& argumentTypes,
     TypePtr returnType) {
-  return std::make_shared<FunctionType>(
+  return std::make_shared<const FunctionType>(
```

Comment:
Is 'const' needed here?

Reply:
Otherwise type in signature and returned type is different and shared_ptr ctor will be called instead of nrvo